### PR TITLE
png support

### PR DESCRIPTION
**Pull request by randaz81**
It is now possible to save image files into .png format

### CODE REVIEW

File: src/libYARP_sig/src/ImageFile.cpp [hunk 1]
```diff
@@ -250,6 +254,93 @@ static bool ImageReadMono(ImageOf<PixelMono> &img, const char *filename)
 // private write methods
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+#if YARP_HAS_PNG_C
+static bool SavePNG(char *src, const char *filename, size_t h, size_t w, size_t rowSize, png_byte color_type, png_byte bit_depth)
+{
```
Comment (Nicogene) [inline src/libYARP_sig/src/ImageFile.cpp]: `src` and `filename` are used but not checked to be `!=nullptr`, I'd like to add this check.

File: src/libYARP_sig/include/yarp/sig/ImageFile.h [hunk 2]
```diff
@@ -29,7 +29,8 @@ namespace yarp {
                     FORMAT_PGM,
                     FORMAT_PPM,
                     FORMAT_JPG,
-                    FORMAT_NUMERIC
+                    FORMAT_NUMERIC,
+                    FORMAT_PNG
                 };
```
Comment (Nicogene) [inline src/libYARP_sig/include/yarp/sig/ImageFile.h]: Is it missing ` bool YARP_sig_API write(const ImageOf<PixelRgb>& src, const std::string& dest, image_fileformat format = FORMAT_PNG);` ?

File: src/libYARP_sig/src/ImageFile.cpp (hunk 1 above)
Comment (drdanz) [reply src/libYARP_sig/src/ImageFile.cpp]: Unfortunately this is not checked in the other similar methods as well. I'd start by adding the check here, and later fix the other methods

File: src/libYARP_sig/include/yarp/sig/ImageFile.h (hunk 2 above)
Comment (drdanz) [reply src/libYARP_sig/include/yarp/sig/ImageFile.h]: I believe this is correct, it is the comment `// PPM/PGM format` next to these methods that is misleading...

**Comment by drdanz** [comment]
Also it would be very nice if you could add in the `.travis-yml` file
* `libpng-dev` (in the `apt` section)
* `libpng` (the homebrew section, but it requires #1909 to be merged)

File: src/libYARP_sig/src/ImageFile.cpp (hunk 1 above)
Comment (randaz81) [reply src/libYARP_sig/src/ImageFile.cpp]: libpng is a C library. Its APIs do not allow neither the use of smart pointers, nor fstream.

**Pull request merged**